app/scripts/directives/fromFile.js

```diff
@@ -170,22 +172,29 @@ angular.module("openshiftConsole")
           });
         };
 
-        // Takes item that will be inspect for kind, metadata fields and if the item is meant to be created in current namespace
-        function validateFields(item) {
+        // Takes item that will be inspect kind field.
+        function validateKind(item) {
```

Comment:
Looks good. Just one nit on the name: I think `isKindValid` and `isMetadataValid` are better names for these functions. The current name sounds like "Should I validate the metadata?" rather than "Is the metadata valid?"